### PR TITLE
unittest for Stream.php

### PR DESCRIPTION
**Pull request by bennija**
There are some test with "//Todo" above them, please tell me how to make those work properly.

### CODE REVIEW

File: test/StreamTest.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    public function testWriteReadAndEof()
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $stream->write('test');
+        $stream->read(4);
+        $eof = $stream->eof();
+        $this -> assertSame(true, $eof);
+    }
+
+    //Todo does this test make any sense
```
Comment (midahp) [inline test/StreamTest.php]: No this does not make any sense. You should seek to a specific position and then read some bytes and assert that the read bytes are the expected ones. You could also use `tell` for that. Also maybe use some longer strings for testing.

File: test/StreamTest.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        $this -> assertSame(null, $seek);
+    }
+
+    public function testRewind()
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $stream->write('test');
+        $stream->read(4);
+        $stream->rewind();
+        $eof = $stream->eof();
```
Comment (midahp) [inline test/StreamTest.php]: You are only checking if you are not at eof after `rewind`, but you want to check if you are at the beginning after `rewind`. For that you can use either `tell` or `read`

File: test/StreamTest.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        $this -> assertSame(false, $eof);
+    }
+
+    //Todo getContents allways returns ''
+    public function testGetContents()
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $stream->write('test');
+        $stream->read(2);
+        $content = $stream->getContents();
```
Comment (midahp) [inline test/StreamTest.php:117]: This test is correct, but a bit longer test string would be helpful here.

File: test/StreamTest.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    //TODO getSize() returns null
+    public function testGetSize()
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $stream->write('test');
+        $size = $stream->getSize();
+        $this -> assertSame(null, $size);
+    }
+
+    public function testTell()
```
Comment (midahp) [inline test/StreamTest.php:75]: Also add a test to make sure that `tell` returns 0 if you call it immediately

File: test/StreamTest.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $isWritable = $stream->isWritable();
+        $this -> assertSame(true, $isWritable);
+    }
+
+    public function testWriteToString()
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $stream->write('test');
```
Comment (midahp) [inline test/StreamTest.php]: This test would also benefit from a longer test string.
You should also add a test that uses a unseekable stream.

File: test/StreamTest.php [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    public function testGetContents()
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $stream->write('test');
+        $stream->read(2);
+        $content = $stream->getContents();
+        $this -> assertSame('st', $content);
+    }
+
+    public function testGetMetadata()
```
Comment (midahp) [inline test/StreamTest.php:121]: Add a test that expects `null` to returned for an invalid key.
Also add a test that does not pass a key and checks the returned array.

**Review by midahp — CHANGES_REQUESTED** [review]
Please try to change as mentioned.
After that you can also try to test the `detach` method

File: test/StreamTest.php [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+{
+    public function setUp(): void
+    {
+    }
+
+    public function testIsSeekable()
+    {
+        $stream = new Stream(fopen('php://temp', 'r'));
+        $isReadable = $stream->isSeekable();
+        $this -> assertSame(true, $isReadable);;
```
Comment (midahp) [inline test/StreamTest.php]: Please remove the spaces around `->`. Also in the rest of the code.

File: test/StreamTest.php [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        $stream->rewind();
+        $rewinded = $stream->read(9);
+        $this->assertSame('123456789', $rewinded);
+    }
+
+    public function testGetContents()
+    {
+        $stream = new Stream(fopen('php://temp', 'r+'));
+        $stream->write('123456789TestXYZ');
+        $stream->read(2);
```
Comment (midahp) [inline test/StreamTest.php]: You want to use `rewind` then `read` and then `getContents`

File: test/StreamTest.php [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+        $content = $stream->getContents();
+        $this->assertSame('', $content);
+    }
+
+    public function testGetMetadata()
```
Comment (midahp) [inline test/StreamTest.php:121]: You can also add a test that uses `detach` before using `getMetadata` and expect `null` or `[]`

**Review by midahp — COMMENTED** [review]
Looks good.
I just added two small comments
